prime-router/src/main/kotlin/tokens/OktaAuthentication.kt

```diff
@@ -2,53 +2,122 @@ package gov.cdc.prime.router.tokens
 import gov.cdc.prime.router.Organization
 import gov.cdc.prime.router.azure.ActionHistory
-import gov.cdc.prime.router.azure.AuthenticatedClaims
-import gov.cdc.prime.router.azure.AuthenticationVerifier
 import gov.cdc.prime.router.azure.HttpUtilities
-import gov.cdc.prime.router.azure.OktaAuthenticationVerifier
-import gov.cdc.prime.router.azure.PrincipalLevel
-import gov.cdc.prime.router.azure.TestAuthenticationVerifier
 import gov.cdc.prime.router.azure.WorkflowEngine
 import gov.cdc.prime.router.common.Environment
 import org.apache.logging.log4j.kotlin.Logging
 
+// These constants match how PRIME Okta subscription is configured
+const val oktaGroupPrefix = "DH"
+const val oktaSenderGroupPrefix = "DHSender_"
+const val oktaAdminGroupSuffix = "Admins"
+const val oktaSystemAdminGroup = "DHPrimeAdmins"
+const val oktaSubjectClaim = "sub"
+const val oktaMembershipClaim = "organization"
+const val envVariableForOktaBaseUrl = "OKTA_baseUrl"
+
+enum class PrincipalLevel {
+    SYSTEM_ADMIN,
+    ORGANIZATION_ADMIN,
+    USER
+}
+
 class OktaAuthentication(private val minimumLevel: PrincipalLevel = PrincipalLevel.USER) : Logging {
-    private val missingAuthorizationHeader = HttpUtilities.errorJson("Missing Authorization Header")
-    private val invalidClaim = HttpUtilities.errorJson("Invalid Authorization Header")
+    private val issuerBaseUrl: String = System.getenv(envVariableForOktaBaseUrl) ?: ""
+
+    companion object : Logging {
+
+        val authenticationFailure = HttpUtilities.errorJson("Authentication Failed")
+        val authorizationFailure = HttpUtilities.errorJson("Unauthorized")
 
-    fun getAccessToken(request: HttpRequestMessage<String?>): String? {
-        // RFC6750 defines the access token
-        val authorization = request.headers[HttpHeaders.AUTHORIZATION.lowercase()] ?: return null
-        return authorization.substringAfter("Bearer ", "")
+        fun getAccessToken(request: HttpRequestMessage<String?>): String? {
+            // RFC6750 defines the access token
+            val authorization = request.headers[HttpHeaders.AUTHORIZATION.lowercase()] ?: return null
+            return authorization.substringAfter("Bearer ", "")
+        }
     }
 
-    companion object : Logging {
-        private var httpRequestMessage: HttpRequestMessage<String?>? = null
+    /**
+     * Perform authentication on a human user.
+     *
+     * Return the claims found in the jwt if the jwt token in [request] is validated.
+     * Return null if not authenticated.
+     * Always performs authentication using Okta, unless running locally.
+     */
+    fun authenticate(request: HttpRequestMessage<String?>): AuthenticatedClaims? {
+        val accessToken = getAccessToken(request)
+        return authenticate(accessToken, request.httpMethod, request.uri.path)
+    }
 
-        fun setRequest(request: HttpRequestMessage<String?>) {
-            httpRequestMessage = request
+    /**
+     * See full comments above.
```

Comment:
This comment will not help in kdoc.  Not sure if there is an @ see tag or something